Revise SL2 requirements for session handling (#120)
openid/ipsie: ipsie-levels.md

| IPSIE<br>LEVEL|   Application (aka RP)                                                 |  Identity Service                                                                                             |
|---------------|----------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------|
| SL1           |   - MUST meet NIST 800-63-4 FAL2 compliance* <br>- Application-specific session lifetime MUST be set from the assertion | - MUST meet NIST 800-63-4 FAL2 Compliance* <br> - MUST enforce MFA and communicate an authentication class to the Application |

> | SL2           |  - When requested from the Identity Service, MUST require the user log in at the Identity Service before continuing any action. <br> MUST revoke any active sessions and tokens at the request of the Identity Service <br> - MUST not accept unsolicited federation assertions| - MUST enforce authentication method requests from Application |

Reviewer: ```suggestion
| SL2           |  - When requested from the Identity Service, MUST require the user to re-authenticate at the Identity Service before continuing any action. <br> MUST revoke any active sessions and tokens at the request of the Identity Service <br> - MUST not accept unsolicited federation assertions| - MUST enforce authentication method requests from Application |
```

Reviewer: "re-authenticate" for consistency

> | SL2           |  - When requested from the Identity Service, MUST require the user to re-authenticate at the Identity Service before continuing any action. <br> MUST revoke any active sessions and tokens at the request of the Identity Service <br> - MUST not accept unsolicited federation assertions| - MUST enforce authentication method requests from Application |

Author: ```suggestion
| SL2           |  - When requested from the Identity Service, MUST obtain a new identity assertion for the user from the Identity Service before continuing any action. <br> MUST revoke any active sessions and tokens at the request of the Identity Service <br> - MUST not accept unsolicited federation assertions| - MUST enforce authentication method requests from Application |
```

Applications MUST NOT accept unsolicited federation requests from the identity service (e.g. SAML IdP initiated federation).

> When requested by the Identity Service, Applications MUST require the specified user log in again at the Identity Service before continuing any action within the Application. This is effectively a "force re-authenticate" command, or an "expire session" command. This command does not place any requirements on what the Application does with artifacts like session identifiers or tokens. Identity Services are not obligated to send this command, but Applications are required to support receiving it.

Reviewer: ```suggestion
When requested by the Identity Service, Applications MUST require the specified user re-authenticate at the Identity Service before continuing any action within the Application. This is effectively a "force re-authenticate" command, or an "expire session" command. This command does not place any requirements on what the Application does with artifacts like session identifiers or tokens. Identity Services are not obligated to send this command, but Applications are required to support receiving it.
```

Reviewer: "re-authenticate" for consistency

> When requested by the Identity Service, Applications MUST require the specified user re-authenticate at the Identity Service before continuing any action within the Application. This is effectively a "force re-authenticate" command, or an "expire session" command. This command does not place any requirements on what the Application does with artifacts like session identifiers or tokens. Identity Services are not obligated to send this command, but Applications are required to support receiving it.

Author: ```suggestion
When requested by the Identity Service, Applications MUST obtain a new identity assertion from the Identity Service for the specified user before continuing any action within the Application. This is effectively a "force re-authenticate" command, or an "re-establish session" command. This command does not place any requirements on what the Application does with artifacts like session identifiers or tokens. Identity Services are not obligated to send this command, but Applications are required to support receiving it.
```
